lib/liquid/block_body.rb

```diff
@@ -71,15 +71,23 @@ def self.raise_missing_variable_terminator(token, parse_context)
     # @api private
     def self.render_node(context, output, node)
       node.render_to_output_buffer(context, output)
-    rescue UndefinedVariable, UndefinedDropMethod, UndefinedFilter => e
-      context.handle_error(e, node.line_number)
-    rescue MemoryError
-      raise
-    rescue ::StandardError => e
-      line_number = node.is_a?(String) ? nil : node.line_number
-      error_message = context.handle_error(e, line_number)
-      if node.instance_of?(Variable) || !node.blank? # conditional for backwards compatibility
-        output << error_message
+    rescue => exc
+      blank_tag = !node.instance_of?(Variable) && node.blank?
+      rescue_render_node(context, output, node.line_number, exc, blank_tag)
```

Comment:
Why not pass the `node` to `rescue_render_node` instead? Seems weird that the name is `rescue_render_node` and it's not receiving the `node`, but instead receiving only some parts.

Reply:
Passing the `node` would require allocating that node in liquid-c, which I am trying to avoid.

Reply:
Ah got it! Missed that you didn't have the node instance in C.